### PR TITLE
profetch: init at 0.1.6

### PR DESCRIPTION
**Pull request by q60**
profetch is a system information fetcher written in GNU/Prolog

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/profetch/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib
+, fetchFromGitHub, gprolog }:
+
+stdenv.mkDerivation rec {
+  pname = "profetch";
+  version = "v0.1.6";
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:6]:
```suggestion
  version = "0.1.6";
```

File: pkgs/tools/misc/profetch/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  src = fetchFromGitHub {
+    owner = "RustemB";
+    repo = pname;
+    rev = version;
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:11]:
```suggestion
    rev = "v${version};
```

File: pkgs/tools/misc/profetch/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "1clh3l50wz6mlrw9kx0wh2bbhnz6bsksyh4ngz7givv4y3g9m702";
+  };
+
+  buildInputs = [ gprolog ];
+
+  buildPhase = ''
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:17]:
```suggestion
  buildPhase = ''
    runHook preBuild
```

File: pkgs/tools/misc/profetch/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    gplc profetch.pl --no-top-level            \
+                     --no-debugger --no-fd-lib \
+                     --no-fd-lib-warn --min-size -o profetch
+  '';
+
+  installPhase = ''
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:23]:
```suggestion
    runHook postBuild
  '';

  installPhase = ''
    runHook preInstall
```

File: pkgs/tools/misc/profetch/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    install -Dm755 -t $out/bin profetch
+  '';
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:25]:
```suggestion
    runHook postInstall
  '';
```

File: pkgs/tools/misc/profetch/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with lib; {
+    description = "System Information Fetcher Written in GNU/Prolog";
+    homepage = "https://github.com/RustemB/profetch";
+    license = licenses.gpl3;
```
Comment (OPNA2608) [inline pkgs/tools/misc/profetch/default.nix:30]: GPL3-only or GPL3+? Please clarify.

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/profetch/default.nix (hunk 6 above)
Comment (q60) [reply pkgs/tools/misc/profetch/default.nix:32]: GPL3-only.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133616` at 90cc1994 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>profetch</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild`.

Near pkgs/tools/misc/profetch/default.nix:16:3:
```
   |
16 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook postInstall`.

Near pkgs/tools/misc/profetch/default.nix:23:3:
```
   |
23 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/profetch/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  version = "0.1.6";
+
+  src = fetchFromGitHub {
+    owner = "RustemB";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1clh3l50wz6mlrw9kx0wh2bbhnz6bsksyh4ngz7givv4y3g9m702";
+  };
+
+  buildInputs = [ gprolog ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/profetch/default.nix:14]:
```suggestion
  nativeBuildInputs = [ gprolog ];
```

File: pkgs/tools/misc/profetch/default.nix [hunk 8]
```diff
@@ -23,6 +23,7 @@ stdenv.mkDerivation rec {
   installPhase = ''
       runHook preInstall
       install -Dm755 -t $out/bin profetch
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/profetch/default.nix]:
```suggestion
    runHook preInstall
    install -Dm755 -t $out/bin profetch
```

**Comment by q60** [comment]
anybody?

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 133616` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>profetch</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]

**Pull request merged**